### PR TITLE
TP-655: Added more tests for the importer system.

### PR DESCRIPTION
**Pull request by NDevox**
Added tests for the nursery caching interface as well as the tasks.

There's still elements of the importer untested, and lots of edge cases which could have done with testing. But no time to do it in!

I'll document some of the key bits that are missing on the ticket.

### CODE REVIEW

File: importer/tests/conftest.py [hunk 1]
```diff
@@ -107,8 +111,33 @@ def prepped_handler_with_dependencies2(
...
+def batch_dependency() -> models.BatchDependencies:
+    dependencies = factories.BatchDependenciesFactory.create()
+    factories.ImporterXMLChunkFactory.create(batch=dependencies.depends_on)
+    factories.ImporterXMLChunkFactory.create(batch=dependencies.dependent_batch)
+    return dependencies
+
+
+@pytest.fixture
+def transaction_element() -> etree.Element:
+    pass
```
Comment (simonwo) [inline importer/tests/conftest.py:143]: What's this for?

**Review by simonwo — COMMENTED** [review]
Looks like a good set.

File: importer/tests/conftest.py (hunk 1 above)
Comment (NDevox) [reply importer/tests/conftest.py:143]: Ah was me starting on the chunker tests before being mildly overwhelmed by them,

File: common/tests/factories.py [hunk 2]
```diff
@@ -933,8 +934,22 @@ class Meta:
...
     class Meta:
         model = "importer.ImporterXMLChunk"
 
     batch = factory.SubFactory(ImportBatchFactory)
+    chunk_number = 1
+    status = ImporterChunkStatus.WAITING
+    chunk_text = """\
+<?xml version="1.0" encoding="UTF-8"?>
+<env:envelope xmlns="urn:publicid:-:DGTAXUD:TARIC:MESSAGE:1.0" xmlns:env="urn:publicid:-:DGTAXUD:GENERAL:ENVELOPE:1.0" id="1">
+</env:envelope>"""
```
Comment (stuaxo) [inline common/tests/factories.py:947]: I tend to use `textwrap.dedent()` to help with this, but I guess there is a little overhead.

**Review by simonwo — APPROVED** [review]
